crypto/src/asset/registry.rs

```diff
@@ -82,24 +82,36 @@ impl Registry {
     /// the parsed display denomination and associated base denomination.
     /// Otherwise, returns `None`.
     pub fn parse_display(&self, raw_denom: &str) -> Option<DisplayDenom> {
-        self.display_set
-            .matches(raw_denom)
-            .iter()
-            .next()
-            .map(|display_index| {
-                let base_index = self.display_to_base[display_index];
-                // We need to determine which unit we matched
-                for (unit_index, regex) in self.display_regexes[base_index].iter().enumerate() {
-                    if let Some(capture) = regex.captures(raw_denom) {
-                        let data = capture.name("data").map(|m| m.as_str()).unwrap_or("");
-                        return DisplayDenom {
-                            inner: Arc::new(self.constructors[base_index](data)),
-                            unit_index,
-                        };
+        if self.parse_base(raw_denom).is_some() {
+            let base_inner = self.parse_base(raw_denom).unwrap().inner;
+
+            // The display version of the base denomination is always the last in the
+            // `units`.
+            let unit_index = base_inner.units.len() - 1;
+            return Some(DisplayDenom {
+                inner: base_inner,
+                unit_index,
+            });
```

Comment:
I think this code could be simpler, by doing something like
```
if let Some(base_denom) = self.parse_base(raw_denom) {
    return Some(base_denom.units().last().expect("base denom always has at least one display denom"));
}
```
since the current code is constructing the whole base denomination twice and then re-constructing the display denom that's already present in it.

Reply:
thanks! this is simpler, updated